File: solutions/csharp/twelve-days/1/TwelveDays.cs
using System;

public static class TwelveDays
{
    public static string Recite(int verseNumber)
    {
        string[] days = {"first","second","third","fourth","fifth","sixth","seventh","eighth","ninth","tenth","eleventh","twelfth"};
        string[] gifts = {"a Partridge in a Pear Tree.","two Turtle Doves,","three French Hens,",
"four Calling Birds,","five Gold Rings,", "six Geese-a-Laying,","seven Swans-a-Swimming,", "eight Maids-a-Milking,", "nine Ladies Dancing,","ten Lords-a-Leaping,", "eleven Pipers Piping,","twelve Drummers Drumming,"};
        string verse = $"On the {days[verseNumber-1]} day of Christmas my true love gave to me:";
        string giftsPhrase = "";
        for (int d = verseNumber - 1 ; d >= 0; d--){
            if (d == 0 && giftsPhrase != "") giftsPhrase += " and";
            giftsPhrase = giftsPhrase + " " + gifts[d];
        }
        verse += giftsPhrase;
        return verse;

        //throw new NotImplementedException("You need to implement this function.");
    }

    public static string Recite(int startVerse, int endVerse)
    {
        string verses = "";
        for (int vn = startVerse; vn <= endVerse; vn++)
        {
            verses += Recite(vn) +'\n';
        }
        return verses.Trim();
        //throw new NotImplementedException("You need to implement this function.");
    }
}

File: solutions/java/twelve-days/1/src/main/java/TwelveDays.java
class TwelveDays {
    String[] days = {"first","second","third","fourth", "fifth","sixth",
                   "seventh", "eighth", "ninth", "tenth", "eleventh", "twelfth"};
    String[] presents = {"a Partridge in a Pear Tree","two Turtle Doves", "three French Hens","four Calling Birds",
                         "five Gold Rings","six Geese-a-Laying","seven Swans-a-Swimming","eight Maids-a-Milking",
                         "nine Ladies Dancing","ten Lords-a-Leaping","eleven Pipers Piping","twelve Drummers Drumming"};

       String verse(int verseNumber) {
        String verse = "On the " + days [verseNumber - 1] + " day of Christmas my true love gave to me, ";
        for(int i = verseNumber - 1; i > 0; i--){
            //System.out.println("i="+i);
            verse += presents[i]+ ", ";
        }
        if (verseNumber > 1)verse += "and ";
        verse += presents[0]+".\n";
        //System.out.println("verse="+verse);
        return(verse);
    }

    String verses(int startVerse, int endVerse) {
           String verse ="";
        for(int vi = startVerse; vi < endVerse; vi++) verse += verse(vi) + "\n";
        verse += verse(endVerse);
        return(verse);
    }
    
    String sing() {
        return(verses(1,12));
    }
}